tools/c7n_gcp/c7n_gcp/resources/sql.py

```diff
@@ -68,3 +68,34 @@ def get_resource_params(self, model, resource):
         return {'project': project,
                 'instance': instance,
                 'body': {'settings': {'activationPolicy': 'NEVER'}}}
+
+
+@resources.register('sql-database')
+class SqlDatabase(ChildResourceManager):
+
+    def _get_parent_resource_info(self, resource_info):
+        return {
+            'name': resource_info['instance'],
```

Comment:
this metadata (how to get parent id from child) should also be in parent_spec, which removes the burden to override this method for the common case.

Reply:
imo, putting this inside the `parent_spec` will make the latter one too cumbersome. Anticipating possible suggestion to stick with jmespath somehow, I would say that it will make the code unreadable forcing a developer to go back and forth guessing what all that stuff means. Using dictionary to bring a little bit of clarification for this mapping (`resource_info` -> `parent_resource_info`) will make it even more cumbersome as the whole `parent_spec` is already dictionary.

Reply:
Continuing discussion about jmespath, I can imagine a situation where simple mapping won't be enough - KMS. There is only one required argument for all the resources - `parent`. And to get that for a parent resource, we have to substring the one for child.

Reply:
just because the literal  key name is `parent` doesn't make it significantly different than other usage of qualified name. ie, its the equivalent to location or zone style, or even qualified name usage that exists for most resources that is handled by get.

Reply:
There is no more burden to override this, but there is such possibility as discussed [below](https://github.com/cloud-custodian/cloud-custodian/pull/3736#issuecomment-479103538) (for those cases like KMS example given above)